MetaMask/metamask-extension
# Replace local code fence transform with `@metamask/build-utils`

## **Description**

In https://github.com/MetaMask/metamask-mobile/pull/7734, code fence support was added to MetaMask Mobile by essentially copypasting the implementation from the extension. This PR replaces the local code fence implementation with a version that was extracted into its own package.

The external package implements code fence parsing and removal, as well as linting transformed files. The consumer is responsible for integrating the fence parser into the given build system, and configuring the ESLint instance used for linting. Consequently, the following remains in this repository after extracting the fence parsers:
- `RemoveFencedCodeTransform`, a transform stream compatible with Browserify
- A utility for configuring and initializing a singleton ESLint instance

The PR is deceptively large; most changes are due to renaming existing fences. For reviewers, the relevant files are:
- `development/build/transforms/*`
- `lavamoat/build-system/policy.json`

## **Related issues**

- https://github.com/MetaMask/core/pull/3577
- https://github.com/MetaMask/metamask-mobile/pull/7972

## **Manual testing steps**

1. Create different builds of the extension and verify that they are working and include the correct features

## **Pre-merge reviewer checklist**
